packages/component_code_gen/templates/generate_apps.py

```diff
@@ -0,0 +1,466 @@
+      },
+    }
+  },
+  methods: {
+    async _makeRequest($ = this, opts) {
+      const {
+        method = "get",
+        path,
+        data,
+        params,
+        ...otherOpts
+      } = opts;
+      return await axios($, {
+        ...otherOpts,
+        method,
+        url: `https://api.the_app_name.com${path}`, // the base URL of the app API
+        headers: {
+          ...opts.headers,
+          "Authorization": `Bearer ${this.$auth.oauth_access_token}`, // the authentication type depends on the app
+        },
+        params,
+        data,
+      })
+    },
+    async getOptions() {
+      // the code to get the options
+      return await this._makeRequest({
+        ...opts,
+      })
+    },
+  },
+}
+```
+
+This lets the user connect their app account to the step, authorizing requests to the app API.
+
+`this` exposes the user's app credentials in the object `this.$auth`. For integrations where users provide static API keys / tokens, the $auth object contains properties for each key / token the user enters. For OAuth integrations, this object exposes the OAuth access token in the oauth_access_token property of the $auth object.
+
+The app can be a key-based app. For integrations where users provide static API keys / tokens, `this.$auth` contains properties for each key / token the user enters. Users are asked to enter custom fields. They are each exposed as properties in the object `this.$auth`. When you make the API request, use the format from the app docs. Different apps pass credentials in different places in the HTTP request, e.g. headers, url params, etc.
+
+The app can also be an OAuth app. For OAuth integrations, this object exposes the OAuth access token in the variable `this.$auth.oauth_access_token`. When you make the API request, make sure to use the format from the app docs, e.g. you may need to pass the OAuth access token as a Bearer token in the Authorization header.
+
+The app code should contain a `propDefinitions` property, which are the definitions for the props. Props lets the user pass data to the step via a form in the Pipedream UI, so they can fill in the values of the variables. Include any required parameters as properties of the `props` object. Props must include a human-readable `label` and a `type` (one of string|boolean|integer|object) that corresponds to the Node.js type of the required param. string, boolean, and integer props allow for arrays of input, and the array types are "string[]", "boolean[]", and "integer[]" respectively. Complex props (like arrays of objects) can be passed as string[] props, and each item of the array can be parsed as JSON. If the user asks you to provide an array of object, ALWAYS provide a `type` of string[]. Optionally, props can have a human-readable `description` describing the param. Optional parameters that correspond to the test code should be declared with `optional: true`. Recall that props may contain an `options` method.
```

Comment:
It looks like the text starting with "Props lets the user pass data to the step via a form in the Pipedream UI" is also shared across actions + sources, so we should define that one place and include it here. I'd like to test changes to text like this in the future to further improve the instructions, so that'll be easier when it's just in one place.